tool/lib/commands/repo_check.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:args/command_runner.dart';
+
+import '../model.dart';
+
+class RepoCheckCommand extends Command {
+  @override
+  String get name => 'repo-check';
+
+  @override
+  String get description => 'Validate properties of the repo source code.';
+
+  @override
+  Future run() async {
+    final repo = DevToolsRepo.getInstance();
+    print('DevTools repo at ${repo.repoPath}.');
+
+    final checks = <Check>[
+      DevToolsVersionCheck(),
+    ];
+
+    print('\nPerforming checks...');
+
+    int failureCount = 0;
+
+    for (var check in checks) {
+      print('\nchecking ${check.name}:');
+
+      try {
+        await check.performCheck(repo);
+
+        print('  check successful');
+      } catch (e) {
+        failureCount++;
+
+        print('  failed: $e');
+      }
+    }
+
+    return failureCount == 0 ? 0 : 1;
+  }
+}
+
+abstract class Check {
+  String get name;
+
+  // Throw if the check fails.
+  Future<void> performCheck(DevToolsRepo repo);
+}
+
+class DevToolsVersionCheck extends Check {
+  @override
+  String get name => 'devtools version';
+
+  @override
+  Future<void> performCheck(DevToolsRepo repo) {
+    final pubspecContents = repo.readFile('packages/devtools_app/pubspec.yaml');
+    final versionString = pubspecContents
+        .split('\n')
+        .firstWhere((line) => line.startsWith('version:'));
+    final pubspecVersion = versionString.substring('version:'.length).trim();
```

Comment:
nit: add a TODO to use a package to parse pubspec files
https://pub.dev/packages/pubspec

Reply:
will do